webhook-service/lib/ip_resolver.go

```diff
@@ -0,0 +1,42 @@
+package lib
+
+import (
+	"net"
+	"net/url"
+
+	logger "github.com/sirupsen/logrus"
+)
+
+type IPResolver interface {
+	Resolve(curlURL string) []string
+}
+
+type LookupFunc func(host string) ([]net.IP, error)
+
+type IpResolver struct {
```

Comment:
you can also just use the unexported name `ipResolver` for this one. Why does it need to be an exported type?
In the tests either change the test package to `lib` or use your constructor function below to create an ipResolver value.